src/delegation/store/traits.rs

```diff
@@ -26,8 +27,12 @@ where
         cid: &Cid,
     ) -> Result<Option<Arc<Delegation<DID, V, C>>>, Self::DelegationStoreError>;
 
-    fn insert(&self, delegation: Delegation<DID, V, C>) -> Result<(), Self::DelegationStoreError> {
-        self.insert_keyed(delegation.cid().expect("FIXME"), delegation)
+    fn insert(
+        &self,
+        delegation: Delegation<DID, V, C>,
+    ) -> Result<(), CannotCidOr<Self::DelegationStoreError>> {
```

Comment:
Would prefer if we rename `Self::DelegationStoreError` to `Error` and `CannotCidOr` to `DelegationStoreError` :v:

However, I'll keep the PR train going and add another PR on top that will do that rename :P